### PR TITLE
DP-2955 - Add examples for auto field mapping and excluded fields

### PR DESCRIPTION
**Pull request by gvjoshi25**

### CODE REVIEW

File: src/main/java/com/northconcepts/datapipeline/foundations/examples/datamapping/MapDataFromSourceToTargetUsingAutoFieldMapping.java [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+
+import com.northconcepts.datapipeline.core.Record;
+import com.northconcepts.datapipeline.foundations.datamapping.DataMapping;
+import com.northconcepts.datapipeline.foundations.datamapping.DataMappingResult;
+import com.northconcepts.datapipeline.internal.expression.DefaultExpressionContext;
+
+public class MapDataFromSourceToTargetUsingAutoFieldMapping {
+
+    public static void main(String[] args) {
+        DefaultExpressionContext input = new DefaultExpressionContext();
```
Comment (DeleTaylor) [inline src/main/java/com/northconcepts/datapipeline/foundations/examples/datamapping/MapDataFromSourceToTargetUsingAutoFieldMapping.java]: Let's use the more common `DataMappingReader` approach in this example.

File: src/main/java/com/northconcepts/datapipeline/foundations/examples/datamapping/MapDataFromSourceToTargetUsingAutoFieldMapping.java [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+ * https://northconcepts.com/data-pipeline/licensing/
+ */
+package com.northconcepts.datapipeline.foundations.examples.datamapping;
+
+import com.northconcepts.datapipeline.core.Record;
+import com.northconcepts.datapipeline.foundations.datamapping.DataMapping;
+import com.northconcepts.datapipeline.foundations.datamapping.DataMappingResult;
+import com.northconcepts.datapipeline.internal.expression.DefaultExpressionContext;
+
+public class MapDataFromSourceToTargetUsingAutoFieldMapping {
```
Comment (DeleTaylor) [inline src/main/java/com/northconcepts/datapipeline/foundations/examples/datamapping/MapDataFromSourceToTargetUsingAutoFieldMapping.java]: Not bad name, but let's use `MapDataUsingAutoMappingAndExcludedFields`.

**Review by DeleTaylor — CHANGES_REQUESTED** [review]

File: src/main/java/com/northconcepts/datapipeline/foundations/examples/datamapping/MapDataFromSourceToTargetUsingAutoFieldMapping.java (hunk 1 above)
Comment (gvjoshi25) [reply src/main/java/com/northconcepts/datapipeline/foundations/examples/datamapping/MapDataFromSourceToTargetUsingAutoFieldMapping.java]: Done

File: src/main/java/com/northconcepts/datapipeline/foundations/examples/datamapping/MapDataFromSourceToTargetUsingAutoFieldMapping.java (hunk 2 above)
Comment (gvjoshi25) [reply src/main/java/com/northconcepts/datapipeline/foundations/examples/datamapping/MapDataFromSourceToTargetUsingAutoFieldMapping.java]: Done.

**Review by DeleTaylor — APPROVED** [review]

**Pull request merged**